massgov/mayflower
# DP-11507 Move richText.js functionality to Drupal

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

- Remove the part to add the hooks for responsive table (ll.8-) in richText.js.
```
    const $table = $el.find("table").not("table table");
    $table.each(function() {
      const $this = $(this);
      let $wrapped;
      // If this table does not have the `ma__table--responsive` class.
      if (!$this.closest('.ma__table--responsive').length) {
        $this
          .wrap('<div class="ma__table--responsive js-ma-responsive-table"><div class="ma__table--responsive__wrapper"></div></div>');
      }
      // At this point there is a `.ma__table--responsive` class,
      // but does this table already posses a responsive table structure.
      else if(!$this.closest('.ma__table--responsive').length) {
        $this
          .wrap('<div class="ma__table--responsive js-ma-responsive-table"><div class="ma__table--responsive__wrapper"></div></div>');
      }

      $wrapped = $this.closest(".ma__table--responsive");
      if (!$wrapped.find("nav.ma__table__horizontal-nav").length) {
        const $tableNav = $('<nav class="ma__table__horizontal-nav"><button class="ma__table__horizontal-nav__left" aria-controls="{{tableID}}"><span class="visually-hidden">Scroll left</span></button><div class="clip-scrollbar"><div class="ma__scroll-indicator"><div class="ma__scroll-indicator--bar" aria-controls="{{tableID}}" role="scrollbar" aria-orientation="horizontal"><div class="ma__scroll-indicator__button"></div></div></div></div><button class="ma__table__horizontal-nav__right" aria-controls="{{tableID}}"><span class="visually-hidden">Scroll right</span></button></nav>');
        $wrapped.prepend($tableNav);
      }
    });
```

- Add missing class setting shown in table.twig, `.ma__table--wide`, in table-responsive.twig.

- Switch the template for table in rte in the JSON files in 05-pages:
  - information-details~responsive-table.json
  - information-details~toc-and-responsive-tables.json

## Related Issue / Ticket

- [DP-11507 \[BE + FE? Dev\] Convert Mayflower Patternlab's richtext.js into a Drupal text filter](https://jira.mass.gov/browse/DP-11507)
- [DP-11507 Move richText.js functionality from Mayflower to Drupal #2974](https://github.com/massgov/mass/pull/2974)

## Steps to Test


1. Go to:
    - http://localhost:3000/?p=atoms-table-responsive
    - http://localhost:3000/?p=pages-information-details-toc-and-responsive-tables.
2. Check the source for below which the above removed part in richText.js adds to `<table>`s in richtext :
```
<div class="ma__table--responsive js-ma-responsive-table">

  <nav class="ma__table__horizontal-nav" style="width: 1838px;">
    <button class="ma__table__horizontal-nav__left" aria-controls="1">
      <span class="visually-hidden">Scroll left</span>
    </button>

    <div class="ma__scroll-indicator" aria-controls="1" role="scrollbar" aria-orientation="horizontal">
      <div class="ma__scroll-indicator__button"></div>
    </div>

    <button class="ma__table__horizontal-nav__right" aria-controls="1">
      <span class="visually-hidden">Scroll right</span>
    </button>
  </nav>
  <div class="ma__table--responsive__wrapper " id="1" tabindex="0" role="region" aria-orientation="horizontal" aria-label="A description of what the table contains">
    <table class="ma__table sticky-enabled" style="margin: 0px; width: 100%;">
```
Since Mayflower content doesn't have rte content. the template should have the hooks.
3. Change the browser window size to see the table is responsive.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
